core/Archive/ArchiveInvalidator.php

```diff
@@ -68,16 +71,26 @@ public function __construct(Model $model, ArchivingStatus $archivingStatus)
         $this->archivingStatus = $archivingStatus;
     }
 
-    public function rememberToInvalidateArchivedReportsLater($idSite, Date $date)
+    public function getAllRememberToInvalidateArchivedReportsLater()
     {
-        // To support multiple transactions at once, look for any other process to have set (and committed)
-        // this report to be invalidated.
-        $key   = $this->buildRememberArchivedReportIdForSiteAndDate($idSite, $date->toString());
-
         // we do not really have to get the value first. we could simply always try to call set() and it would update or
         // insert the record if needed but we do not want to lock the table (especially since there are still some
         // MyISAM installations)
-        $value = Option::getLike($key . '%');
+        $values = Option::getLike($this->rememberArchivedReportIdStart . '%');
+
+        $all = [];
+        foreach ($values as $name => $value) {
+            $suffix = substr($name, strlen($this->rememberArchivedReportIdStart));
+            list($idSite, $dateStr) = explode('_', $suffix);
+
+            $all[$idSite][$dateStr][$name] = $value;
```

Comment:
Probably not too important but `[$name]` could be removed? It doesn't look at this one in the cached tracker value and it would also prevent that when 200 tracking requests at the same time use the `option::set` that we have 200 keys in there for the same idSite,dateStr combination